Gems/Multiplayer/Code/Source/Debug/MultiplayerConnectionViewportMessageSystemComponent.cpp

```diff
@@ -112,61 +112,75 @@ namespace Multiplayer
         if (agentType == MultiplayerAgentType::Client)
         {        
             // Display the connection status in the bottom-right viewport
+            m_drawParams.m_hAlign = AzFramework::TextHorizontalAlignment::Right;
+            m_drawParams.m_position = AZ::Vector3(static_cast<float>(viewportSize.m_width), static_cast<float>(viewportSize.m_height), 1.0f) + AZ::Vector3(viewportConnectionBottomRightBorderPadding) * viewport->GetDpiScalingFactor();
+
             if (AzNetworking::INetworkInterface* networkInterface = AZ::Interface<AzNetworking::INetworking>::Get()->RetrieveNetworkInterface(AZ::Name(MpNetworkInterfaceName)))
             {
                 AzNetworking::IConnectionSet& connectionSet = networkInterface->GetConnectionSet();
                 if (connectionSet.GetConnectionCount() > 0)
                 {
-                    connectionSet.VisitConnections([this](AzNetworking::IConnection& connection){ this->DrawConnectionStatus(connection.GetConnectionState()); });
+                    connectionSet.VisitConnections([this](AzNetworking::IConnection& connection)
+                    {
+                        m_hostIpAddress = connection.GetRemoteAddress();
+                        this->DrawConnectionStatus(connection.GetConnectionState(), m_hostIpAddress);
```

Comment:
Consider checking ConnectionState and ConnectionRole here to try and filter this down in the event there's multiple connections.

Reply:
I think running on the client every connection role will be the "acceptor" since the client will always be the "connector". So in the case of multiple connections the client would see each machine it's connected.

Reply:
In that case, you could try offsetting subsequent connections vertically so they don't stack up to a certain limit (maybe 8 max or something).

Reply:
Sweet! Added cl_viewportConnectionStatusMaxDrawCount!